typescript/deploy-sdk/src/AltVMHookModule.ts

```diff
@@ -1,88 +1,69 @@
-import { ChainMetadataManager } from '../metadata/ChainMetadataManager.js';
-import { MultiProvider } from '../providers/MultiProvider.js';
-import {
-  AnnotatedTypedTransaction,
-  ProtocolReceipt,
-} from '../providers/ProviderType.js';
-import { ChainName, ChainNameOrId } from '../types.js';
-import { normalizeConfig } from '../utils/ism.js';
-
 import { AltVMHookReader } from './AltVMHookReader.js';
-import {
-  HookConfig,
-  HookConfigSchema,
-  HookType,
-  IgpHookConfig,
-  MUTABLE_HOOK_TYPE,
-} from './types.js';
-
-type HookModuleAddresses = {
-  deployedHook: Address;
-  mailbox: Address;
-};
 
-export class AltVMHookModule<PT extends ProtocolType> extends HyperlaneModule<
-  PT,
-  HookConfig,
-  HookModuleAddresses
-> {
+export class AltVMHookModule implements HypModule<HookModuleType> {
   protected readonly logger = rootLogger.child({
     module: 'AltVMHookModule',
   });
   protected readonly reader: AltVMHookReader;
 
-  // Adding these to reduce how often we need to grab from ChainMetadataManager.
-  public readonly chain: ChainName;
-  public readonly chainId: ChainId;
-  public readonly domainId: Domain;
+  // Cached chain name
+  public readonly chain: string;
 
   constructor(
-    protected readonly metadataManager: ChainMetadataManager,
-    params: HyperlaneModuleParams<HookConfig, HookModuleAddresses>,
-    protected readonly signer: AltVM.ISigner<
-      AnnotatedTypedTransaction<PT>,
-      ProtocolReceipt<PT>
-    >,
+    protected readonly chainLookup: ChainLookup,
+    private readonly args: HypModuleArgs<HookModuleType>,
+    protected readonly signer: AltVM.ISigner<AnnotatedTx, TxReceipt>,
   ) {
-    params.config = HookConfigSchema.parse(params.config);
-    super(params);
+    // this.args.config = HookConfigSchema.parse(this.args.config);
 
-    this.reader = new AltVMHookReader(metadataManager, signer);
+    this.reader = new AltVMHookReader(chainLookup.getChainMetadata, signer);
 
-    this.chain = metadataManager.getChainName(this.args.chain);
-    this.chainId = metadataManager.getChainId(this.chain);
-    this.domainId = metadataManager.getDomainId(this.chain);
+    const metadata = chainLookup.getChainMetadata(this.args.chain);
+    this.chain = metadata.name;
   }
 
-  public async read(): Promise<HookConfig> {
+  public async read(): Promise<DerivedHookConfig> {
     return this.reader.deriveHookConfig(this.args.addresses.deployedHook);
   }
 
+  public serialize(): HookModuleAddresses {
+    return this.args.addresses;
+  }
+
   public async update(
-    targetConfig: HookConfig,
-  ): Promise<AnnotatedTypedTransaction<PT>[]> {
-    if (targetConfig === zeroAddress) {
+    targetConfig: HookConfig | Address,
+  ): Promise<AnnotatedTx[]> {
+    if (typeof targetConfig === 'string' && isZeroishAddress(targetConfig)) {
       return Promise.resolve([]);
     }
 
-    targetConfig = HookConfigSchema.parse(targetConfig);
+    // targetConfig = HookConfigSchema.parse(targetConfig);
```

Comment:
Should we keep this commented here as a reminder?

Reply:
No, removing